base-ci-resource.yaml

```diff
@@ -0,0 +1,192 @@
+
+# Default before script 
+before_script:
+  - export VERSION=${CI_COMMIT_SHA:0:7}
+
+#---Build and testing jobs---#
+
+# If project is not a node app it will use this job configuration for build and testing 
+testNodeApplication:
+  extends: .node
+  image: briangweber/docker-node:carbon
+  stage: build  
+  only:
+    variables:       
+      - $NODEPOSTGRESAPP == null  
+  script:
+    - npm test
+  coverage: '/All files\s*\|\s*[.?\d]+\s*\|\s*([0-9.]+)/'
+  artifacts:
+    paths:
+      - coverage/
+
+# if project is a node application using a postgres db this will be the build and test job 
+testNodeApplicationUsingPostgres:
+  extends: .node
+  image: kevinrambaud/node-flyway:8-5.1.4-alpine
+  stage: build  
+  only:
+    variables:       
+      - $NODEPOSTGRESAPP == "true"  
+  script:
+    - cp -r $CI_PROJECT_DIR/sql/* /flyway/sql/
+    - sh /flyway/flyway -url=jdbc:postgresql://$POSTGRES_SERVER:5432/$POSTGRES_DB -schemas=public -user=$POSTGRES_USER -password=$POSTGRES_PASSWORD -placeholders.ENVIRONMENT=$DEPLOYENV migrate
+    - npm test
+  coverage: '/All files\s*\|\s*[.?\d]+\s*\|\s*([0-9.]+)/'
+  artifacts:
+    paths:
+      - coverage/
+  services:
+    - $POSTGRES_VERSION
+  variables:
+    POSTGRES_DB: postgres
+    POSTGRES_USER: postgres
+    POSTGRES_SERVER: postgres
+    POSTGRES_PASSWORD: ''
+
+# for node applications will scan your dependencies for issues at build time. 
+dependency_scanning:
+  extends: .node
+  image: briangweber/docker-node:carbon
+  stage: build
+  only:
+    variables: 
+      - $PLATFORM == "node"
+  allow_failure: true
+  services:
+    - docker:stable-dind
+  script:
+    - export SP_VERSION=$(echo "$CI_SERVER_VERSION" | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1-\2-stable/')
+    - docker run
+        --env DEP_SCAN_DISABLE_REMOTE_CHECKS="${DEP_SCAN_DISABLE_REMOTE_CHECKS:-false}"
+        --volume "$PWD:/code"
+        --volume /var/run/docker.sock:/var/run/docker.sock
+        "registry.gitlab.com/gitlab-org/security-products/dependency-scanning:$SP_VERSION" /code
+  artifacts:
+    paths: [gl-dependency-scanning-report.json]  
+
+#---Publish Jobs---
+
+# job used for publishing the service 
+publish:
+  extends: .node 
+  image: briangweber/docker-node:carbon
+  stage: publish
+  script:
+    - grunt bundle --packageVersion=$VERSION --packageName=$CI_PROJECT_NAME
+  only:
+    - master
+  except:
+    - schedules
+
+#---Migration and Deploy Jobs---
+
+# In database applicaitons/services preforms the migrations job required before deployment in int
+# triggers automatically after publish. 
+migrate-postgres-int:
+  extends: .migrate
+  stage: migrate-int
+  only:
+    refs:
+      - master
+    variables: 
+      - $INT_POSTGRES_SERVER 
+  except:
+    - schedules
+  variables:
+    POSTGRES_SERVER: $INT_POSTGRES_SERVER
+    POSTGRES_USER: $INT_POSTGRES_USER
+    POSTGRES_DB: $INT_POSTGRES_DB
+    # found in gitlab settings
```

Comment:
These are all "found" in gitlab settings, right?

Reply:
Maybe "Set" is better than "found". Or "these should be set...". Something directing the user to do it.